imgui-SFML.cpp

```diff
@@ -1403,4 +1676,183 @@ void updateMouseCursor(sf::Window& window) {
     }
 }
 
+
+#ifdef VIEWPORTS_ENABLE
+
+void SFML_CreateWindow(ImGuiViewport* viewport) {
+#if SFML_VERSION_MAJOR >= 3
+    sf::RenderWindow* window = new sf::RenderWindow(sf::VideoMode({(unsigned)viewport->Size.x,
+                                                                   (unsigned)viewport->Size.y}),
+                                                    "", sf::Style::None);
+#else
+    sf::RenderWindow* window =
+        new sf::RenderWindow(sf::VideoMode(viewport->Size.x, viewport->Size.y), "",
+                             sf::Style::None);
+#endif
+    window->setVisible(false);
+    WindowContext* data = IM_NEW(WindowContext)(window, ImGui::GetCurrentContext(), true);
+    data->windowHasFocus = true;
+    viewport->PlatformUserData = data;
+    viewport->PlatformHandle = window;
+}
+
+void SFML_DestroyWindow(ImGuiViewport* viewport) {
+    if (WindowContext* wc = (WindowContext*)viewport->PlatformUserData) {
+        if (!wc->imContextOwner) {
+            wc->window->close();
+            IM_DELETE(wc);
+        }
+    }
+    viewport->PlatformUserData = nullptr;
+    viewport->PlatformHandle = nullptr;
+}
+
+void SFML_ShowWindow(ImGuiViewport* viewport) {
+    WindowContext* wc = (WindowContext*)viewport->PlatformUserData;
+    wc->window->setVisible(true);
+}
+
+void SFML_SetWindowPos(ImGuiViewport* viewport, ImVec2 pos) {
+    WindowContext* wc = (WindowContext*)viewport->PlatformUserData;
+    wc->window->setPosition(pos);
+}
+
+ImVec2 SFML_GetWindowPos(ImGuiViewport* viewport) {
```

Comment:
Why doesn't sf::Window::getPosition not work here for all platforms?

Reply:
On Windows SFML `sf::Window::getPosition()` returns top left corner position in screen coords including title bar. ImGui requires position of client area (area where stuff is drawn), therefore if value returned by `sf::Window::getPosition()` is used, main viewport becomes offsetted in reference to `io.MousePos` by title bar height.